{"variants":[{"paths":["\/documentation\/dxfeedframework\/promise\/completeexceptionally(_:)"],"traits":[{"interfaceLanguage":"swift"}]}],"hierarchy":{"paths":[["doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework","doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise"]]},"kind":"symbol","abstract":[{"type":"text","text":"Completes computation exceptionally with a specified exception."},{"type":"text","text":" "},{"type":"text","text":"This method does nothing if computation has already "},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()"},{"text":" completed,","type":"text"},{"text":" ","type":"text"},{"type":"text","text":"otherwise "},{"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()","isActive":true},{"type":"text","text":" will return the specified exception."}],"metadata":{"title":"completeExceptionally(_:)","role":"symbol","externalID":"s:15DXFeedFramework7PromiseC21completeExceptionallyyyAA14GraalExceptionOKF","roleHeading":"Instance Method","symbolKind":"method","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"completeExceptionally","kind":"identifier"},{"text":"(","kind":"text"},{"text":"GraalException","kind":"typeIdentifier","preciseIdentifier":"s:15DXFeedFramework14GraalExceptionO"},{"kind":"text","text":") "},{"text":"throws","kind":"keyword"}],"modules":[{"name":"DXFeedFramework"}]},"sections":[],"schemaVersion":{"minor":3,"major":0,"patch":0},"identifier":{"url":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/completeExceptionally(_:)","interfaceLanguage":"swift"},"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"kind":"identifier","text":"completeExceptionally"},{"text":"(","kind":"text"},{"text":"_","kind":"externalParam"},{"text":" ","kind":"text"},{"text":"exception","kind":"internalParam"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:15DXFeedFramework14GraalExceptionO","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/GraalException","text":"GraalException"},{"text":") ","kind":"text"},{"kind":"keyword","text":"throws"}],"languages":["swift"],"platforms":["macOS"]}]},{"kind":"content","content":[{"level":2,"anchor":"discussion","type":"heading","text":"Discussion"},{"type":"unorderedList","items":[{"content":[{"inlineContent":[{"type":"text","text":"Throws : GraalException. Rethrows exception from Java"}],"type":"paragraph"}]}]}]}],"references":{"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/completeExceptionally(_:)":{"title":"completeExceptionally(_:)","abstract":[{"text":"Completes computation exceptionally with a specified exception.","type":"text"},{"text":" ","type":"text"},{"type":"text","text":"This method does nothing if computation has already "},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()"},{"type":"text","text":" completed,"},{"type":"text","text":" "},{"type":"text","text":"otherwise "},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()","isActive":true,"type":"reference"},{"type":"text","text":" will return the specified exception."}],"fragments":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"completeExceptionally"},{"text":"(","kind":"text"},{"text":"GraalException","kind":"typeIdentifier","preciseIdentifier":"s:15DXFeedFramework14GraalExceptionO"},{"text":") ","kind":"text"},{"kind":"keyword","text":"throws"}],"type":"topic","role":"symbol","url":"\/documentation\/dxfeedframework\/promise\/completeexceptionally(_:)","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/completeExceptionally(_:)","kind":"symbol"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/complete(result:)":{"title":"complete(result:)","abstract":[{"type":"text","text":"Completes computation normally with a specified result."},{"type":"text","text":" "},{"text":"This method does nothing if computation has already ","type":"text"},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","isActive":true,"type":"reference"},{"type":"text","text":" completed"},{"type":"text","text":" "},{"type":"text","text":"(normally, exceptionally, or was cancelled),"}],"fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"text":"complete","kind":"identifier"},{"text":"(","kind":"text"},{"text":"result","kind":"externalParam"},{"kind":"text","text":": "},{"preciseIdentifier":"s:15DXFeedFramework11MarketEventC","text":"MarketEvent","kind":"typeIdentifier"},{"text":") ","kind":"text"},{"text":"throws","kind":"keyword"}],"type":"topic","role":"symbol","url":"\/documentation\/dxfeedframework\/promise\/complete(result:)","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/complete(result:)","kind":"symbol"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/cancel()":{"title":"cancel()","abstract":[{"text":"Cancels computation. This method does nothing if computation has already ","type":"text"},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","type":"reference","isActive":true},{"text":" completed.","type":"text"}],"fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"cancel","kind":"identifier"},{"text":"()","kind":"text"}],"type":"topic","role":"symbol","url":"\/documentation\/dxfeedframework\/promise\/cancel()","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()","kind":"symbol"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/isDone()":{"title":"isDone()","abstract":[{"type":"text","text":"Returns "},{"type":"strong","inlineContent":[{"type":"text","text":"true"}]},{"type":"text","text":" when computation has"},{"type":"text","text":" "},{"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/complete(result:)","isActive":true},{"type":"text","text":" completed normally,"},{"type":"text","text":" "},{"text":"or ","type":"text"},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/completeExceptionally(_:)","type":"reference","isActive":true},{"type":"text","text":" exceptionally,"},{"type":"text","text":" "},{"type":"text","text":"or was "},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()"},{"text":" canceled","type":"text"}],"fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"isDone","kind":"identifier"},{"text":"() -> ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:Sb","text":"Bool"}],"type":"topic","role":"symbol","url":"\/documentation\/dxfeedframework\/promise\/isdone()","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","kind":"symbol"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/getException()":{"url":"\/documentation\/dxfeedframework\/promise\/getexception()","type":"topic","role":"symbol","abstract":[{"text":"Returns exceptional outcome of computation. If computation has no  ","type":"text"},{"type":"reference","isActive":true,"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/hasException()"},{"text":" exception,","type":"text"},{"type":"text","text":" "},{"type":"text","text":"then this method returns nil. If computation has completed exceptionally or was cancelled, then"},{"text":" ","type":"text"},{"text":"the result of this method is not nil.","type":"text"},{"type":"text","text":" "},{"text":"If computation was ","type":"text"},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isCancelled()","isActive":true,"type":"reference"},{"type":"text","text":" cancelled, then this method returns an"},{"text":" ","type":"text"},{"text":"instance of GraalException.","type":"text"}],"title":"getException()","kind":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"getException","kind":"identifier"},{"kind":"text","text":"() -> "},{"preciseIdentifier":"s:15DXFeedFramework14GraalExceptionO","text":"GraalException","kind":"typeIdentifier"},{"kind":"text","text":"?"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework":{"kind":"symbol","type":"topic","title":"DXFeedFramework","abstract":[],"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework","url":"\/documentation\/dxfeedframework","role":"collection"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/hasException()":{"title":"hasException()","role":"symbol","abstract":[{"text":"Returns ","type":"text"},{"inlineContent":[{"text":"true","type":"text"}],"type":"strong"},{"type":"text","text":" when computation has completed exceptionally or was cancelled."},{"type":"text","text":" "},{"text":"Use ","type":"text"},{"type":"reference","isActive":true,"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()"},{"text":" method to get the exceptional outcome of the computation.","type":"text"}],"url":"\/documentation\/dxfeedframework\/promise\/hasexception()","kind":"symbol","type":"topic","fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"text":"hasException","kind":"identifier"},{"text":"() -> ","kind":"text"},{"text":"Bool","preciseIdentifier":"s:Sb","kind":"typeIdentifier"}],"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/hasException()"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/isCancelled()":{"title":"isCancelled()","role":"symbol","url":"\/documentation\/dxfeedframework\/promise\/iscancelled()","abstract":[{"text":"Returns ","type":"text"},{"type":"strong","inlineContent":[{"type":"text","text":"true"}]},{"text":" when computation was canceled.","type":"text"},{"type":"text","text":" "},{"text":"Use ","type":"text"},{"type":"reference","isActive":true,"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()"},{"text":" method to get the corresponding exception.","type":"text"}],"kind":"symbol","type":"topic","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"text":"isCancelled","kind":"identifier"},{"text":"() -> ","kind":"text"},{"text":"Bool","kind":"typeIdentifier","preciseIdentifier":"s:Sb"}],"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isCancelled()"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise":{"role":"symbol","type":"topic","title":"Promise","fragments":[{"kind":"keyword","text":"class"},{"text":" ","kind":"text"},{"text":"Promise","kind":"identifier"}],"url":"\/documentation\/dxfeedframework\/promise","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise","kind":"symbol","abstract":[{"type":"text","text":"Result of a computation that will be completed normally or exceptionally in the future."},{"type":"text","text":" "},{"text":"This class is designed to represent a promise to deliver certain result.","type":"text"}],"navigatorTitle":[{"text":"Promise","kind":"identifier"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/GraalException":{"navigatorTitle":[{"text":"GraalException","kind":"identifier"}],"abstract":[{"text":"Represents errors that occur when calling graal function.","type":"text"}],"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/GraalException","kind":"symbol","url":"\/documentation\/dxfeedframework\/graalexception","type":"topic","role":"symbol","fragments":[{"text":"enum","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"GraalException"}],"title":"GraalException"}}}